session/session.go

```diff
@@ -509,6 +510,27 @@ func (s *session) doCommit(ctx context.Context) error {
 			s.GetSessionVars().TxnCtx.IsExplicit && s.GetSessionVars().GuaranteeLinearizability)
 	}
 
+	// Filter out the temporary table key-values.
```

Comment:
I think it's safer to create a MemBuffer implementation that separates temporary table data and transaction data.

Reply:
We can reserve an id range dedicated for temporary table, for example [MaxInt64, maxUint64].
The allocator can be a session local field.
So the mem buffer can detect if this record is for a temp table.

Reply:
The read process would become complex, and handle statement rollback for two different buffer is extremely complex.
We can use different allocator for auto id, but not for table IDs.